datafusion/src/physical_plan/metrics/aggregated.rs

```diff
@@ -35,25 +35,31 @@ pub struct AggregatedMetricsSet {
     final_: Arc<std::sync::Mutex<Vec<ExecutionPlanMetricsSet>>>,
 }
 
+impl Default for AggregatedMetricsSet {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl AggregatedMetricsSet {
     /// Create a new aggregated set
-    pub(crate) fn new() -> Self {
+    pub fn new() -> Self {
```

Comment:
To create a memory-managed version of [`ShuffleWriter`](https://github.com/blaze-init/blaze-rs/blob/master/datafusion-ext/src/shuffle_writer_exec.rs#L457-L508), we need to expose these four APIs for dependent crate usage. 
I'm not sure this should be in its own PR, or I can have it here since it is also metric-related. I can remove this if needed.

Reply:
I think it is fine to be in this PR